catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/XstreamPathConverter.java

```diff
@@ -164,13 +166,11 @@ protected boolean doBasicPathsMatch(final Path pathObj1, final Path pathObj2) {
      */
     private boolean endsMatch(char[] path1, int index1, char[] path2, int index2) {
```

Comment:
iirc that "simplify and then compare those strings" was the first approach implemented, but the difference in runtime still actually ended up being pretty significant in that you go from essentially two traversals of each to a single traversal of each string in the current approach.